ads/ads.go

```diff
@@ -255,3 +257,24 @@ func LookupStreamTypeByRPCMethod(rpcMethod string) (StreamType, bool) {
 		return UnknownStreamType, false
 	}
 }
+
+// ParseRemainingChunksFromNonce checks whether the Diderot server implementation chunked the delta
+// responses because not all resources could fit in the same response without going over the default
+// max gRPC message size of 4MB. A nonce from Diderot always starts with the 64-bit nanosecond
+// timestamp of when the response was generated on the server. Then the number of remaining chunks as
+// a 32-bit integer. The sequence of integers is binary encoded with [binary.BigEndian] then hex
+// encoded. If the given nonce does not match the expected format, this function simply returns 0, as
```

Comment:
Should this return `-1` instead of `0` or some other special value if the nonce didn't match the expected format? It could also return `remainingChunks, err` if that would be more idiomatic 

Otherwise, someone using `ParseRemainingChunksFromNonce` would be unable to distinguish "this is the last chunk in the response" from "the nonce didn't match the format so we don't know how many chunks remain in the response", right?

Reply:
That's kind of the point actually. By default, the behavior in xDS is to not wait for chunks, since that notion doesn't really exist. Ultimately, if the nonce doesn't have the right format, then the only valid behavior is not to wait, that's why it just returns 0. There's literally nothing a caller of this function could do.

Reply:
Zoab brought up a good point offline which is that returning an error at least signals to callers that the nonce wasn't in the right format, and even though they can't do anything about it, they can still log a warning.